src/overlays/actors/ovl_En_Kaizoku/z_en_kaizoku.c

```diff
@@ -63,120 +133,1981 @@ ActorInit En_Kaizoku_InitVars = {
+        Actor_Kill(&this->picto.actor);
+        return;
+    }
+
+    this->unk_3D4.dim.radius = 20;
+    this->unk_3D4.dim.height = 65;
+    this->unk_3D4.dim.yShift = 0;
+    this->unk_2D6 = this->picto.actor.cutscene;
+    this->picto.actor.world.pos.y = player->actor.world.pos.y + 160.0f;
+    this->picto.validationFunc = EnKaizoku_ValidatePictograph;
+    this->picto.actor.flags |= ACTOR_FLAG_400;
+    func_80B85F48(this);
+}
+
+void EnKaizoku_Destroy(Actor* thisx, PlayState* play) {
+    EnKaizoku* this = THIS;
+
+    Effect_Destroy(play, this->unk_3D0);
+    Collider_DestroyCylinder(play, &this->unk_3D4);
+    Collider_DestroyQuad(play, &this->unk_420);
+    Audio_RestorePrevBgm();
+}
+
+s32 func_80B85858(EnKaizoku* this, PlayState* play) {
+    Actor* itemAction;
+    Player* player = GET_PLAYER(play);
+    s16 sp1E = this->unk_2B0;
+
+    for (itemAction = play->actorCtx.actorLists[ACTORCAT_ITEMACTION].first; itemAction != NULL;
+         itemAction = itemAction->next) {
+        //! FAKE:
+        if (1) {}
+        if ((itemAction->id == ACTOR_ARMS_HOOK || itemAction->id == ACTOR_EN_ARROW) && player->unk_D57) {
+            if ((itemAction->id != ACTOR_ARMS_HOOK || this->unk_2B0 != 13) && this->unk_2B0 != 6) {
+                func_80B87C7C(this);
+            }
+            break;
+        }
+    }
+
+    if (sp1E != this->unk_2B0) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void func_80B85900(EnKaizoku* this) {
+    if (this->unk_2B0 == 1) {
+        this->unk_57A = Math_SinS(this->unk_2B4 * 0x1068) * 8920.0f;
+    } else if (this->unk_2B0 != 13) {
+        if ((this->unk_2B0 == 9) || (this->unk_2B0 == 11)) {
+            Math_SmoothStepToS(&this->unk_57A, this->picto.actor.yawTowardsPlayer - this->picto.actor.shape.rot.y, 1,
+                               500, 0);
+            this->unk_57A = CLAMP(this->unk_57A, -0x256F, 0x256F);
+        } else {
+            this->unk_57A = 0;
+        }
+    }
+}
+
+s32 func_80B85A00(EnKaizoku* this, PlayState* play, s16 arg2) {
+    Player* player = GET_PLAYER(play);
+    s16 sp32 = ABS_ALT(this->picto.actor.wallYaw - this->picto.actor.shape.rot.y);
+    s16 sp30 = ABS_ALT(this->picto.actor.yawTowardsPlayer - this->picto.actor.shape.rot.y);
+    Actor* explosiveActor;
+
+    if (func_800BE184(play, &this->picto.actor, 100.0f, 0x2710, 0x4000, this->picto.actor.shape.rot.y)) {
+        if (player->meleeWeaponAnimation == PLAYER_MWA_JUMPSLASH_START) {
+            if (this->unk_2B0 != 2) {
+                func_80B874D8(this, play);
+            }
+            return true;
+        } else {
+            func_80B87900(this);
+            return true;
+        }
+    } else if (func_800BE184(play, &this->picto.actor, 100.0f, 0x5DC0, 0x2AA8, this->picto.actor.shape.rot.y)) {
+        this->picto.actor.shape.rot.y = this->picto.actor.world.rot.y = this->picto.actor.yawTowardsPlayer;
+        if ((this->picto.actor.bgCheckFlags & 8) && (ABS_ALT(sp32) < 12000) &&
```

Comment:
```suggestion
        if ((this->picto.actor.bgCheckFlags & 8) && (ABS_ALT(sp32) < 0x2EE0) &&
```
Both sp32 and sp30 look to be angles of some kind.